service/storage_service.cc

```diff
@@ -187,12 +187,24 @@ std::optional<gms::inet_address> storage_service::get_replace_address() {
 }
 
 bool storage_service::is_replacing() {
-    sstring replace_address_first_boot = _db.local().get_config().replace_address_first_boot();
-    if (!replace_address_first_boot.empty() && _sys_ks.local().bootstrap_complete()) {
-        slogger.info("Replace address on first boot requested; this node is already bootstrapped");
-        return false;
+    const auto& cfg = _db.local().get_config();
+    if (!cfg.replace_node_first_boot().empty()) {
+        if (_sys_ks.local().bootstrap_complete()) {
+            slogger.info("Replace node on first boot requested; this node is already bootstrapped");
+            return false;
+        }
+        return true;
+    }
+
+    if (!cfg.replace_address_first_boot().empty() || !cfg.replace_address().empty()) {
```

Comment:
Same as in init.cc:
1) easier path to removing support for the legacy config options
2) simpler logic overall IMO since to control the error message we'll need to check again which of the cfg options is given.

Reply:
This way when we drop the replace_address_first_boot option, we can drop the old code without touching the new code.  It is also either to reason about the code since less conditions in a single if.

Reply:
I will combine them in the next version, as I realized that `cfg.replace_address().empty()` should be handled separately to follow the existing the logic (and cause an exception down the road if `!cfg.replace_address().empty()` and `_sys_ks.local().bootstrap_complete()`)

Reply:
> This way when we drop the replace_address_first_boot option, we can drop the old code without touching the new code. It is also either to reason about the code since less conditions in a single if.

You managed to convince me to flip my mind again :)